src/EscapeAnalysis.jl

```diff
@@ -133,67 +137,103 @@ end
 # ========
 
 """
-    abstract type EscapeInformation end
-
-A lattice for escape information, which has the following elements:
-- `NoInformation`: the top element of this lattice, meaning no information is derived
-- `NoEscape`: the second topmost element of this lattice, meaning it will not escape from this local frame
-- `ReturnEscape`: a lattice that is lower than `NoEscape`, meaning it will escape to the caller
-- `Escape`: the bottom element of this lattice, meaning it will escape to somewhere
+    abstract type EscapeLattice end
+
+A lattice for escape information, which holds the following properties:
+- `Analyzed`: not formally part of the lattice, indicates this statement has not been analyzed at all
+- `ReturnEscape`: indicates it will escape to the caller via return (possibly as a field)
+- `ArgEscape`: indicates it will escape to the caller through setfield on argument(s)
+  -1 : no escape
+   0 : unknown or multiple
+   n : through argument N
+
+These attributes can be combined to create a partial lattice:
+(note that this is inverted from the order used for the lattice in Core.Compiler)
+- `NoEscape`: the topmost element of this lattice
+- `Escape`: the inverse of NoEscape
```

Comment:
```suggestion
- `Escape`: the inverse of `NoEscape`
```

And also, the "inverse" of `NoEscape` is `AllEscape`, right ?